dev-tools/tests/test_omicron_dev.rs

```diff
@@ -388,25 +417,35 @@ async fn test_run_all() {
         Exec::cmd("bash").arg("-c").arg(cmdstr).stderr(Redirection::Merge);
     let runall = run_run_all(exec);
 
-    // Make sure we can connect to CockroachDB.
-    let (client, connection) = runall
-        .postgres_config
-        .connect(tokio_postgres::NoTls)
-        .await
-        .expect("failed to connect to newly setup database");
-    let conn_task = tokio::spawn(connection);
-    assert!(has_omicron_schema(&client).await);
-    drop(client);
-    conn_task
-        .await
-        .expect("failed to join on connection")
-        .expect("connection failed with an error");
-    eprintln!("cleaned up connection");
-
-    // Make sure we can connect to Nexus.
-    let client =
-        oxide_client::Client::new(&format!("http://{}", runall.external_url));
-    let _ = client.logout().send().await.unwrap();
+    let test_task = async {
+        // Make sure we can connect to CockroachDB.
+        let (client, connection) = runall
+            .postgres_config
+            .connect(tokio_postgres::NoTls)
+            .await
+            .context("failed to connect to newly setup database")?;
+        let conn_task = tokio::spawn(connection);
+        anyhow::ensure!(
+            has_omicron_schema(&client).await,
+            "Missing Omicron schema"
+        );
+        drop(client);
+        conn_task
+            .await
+            .context("failed to join on connection")?
+            .context("connection failed with an error")?;
+        eprintln!("cleaned up connection");
+
+        // Make sure we can connect to Nexus.
+        let client = oxide_client::Client::new(&format!(
+            "http://{}",
+            runall.external_url
+        ));
+        let _ =
+            client.logout().send().await.context("Nexus failed to start")?;
```

Comment:
```suggestion
            client.logout().send().await.context("unexpectedly failed to reach Nexus logout endpoint")?;
```

(we don't know why this failed -- Nexus may well have started)

Reply:
True, fixed.